src/components/app-field.js

```diff
@@ -1,64 +1,157 @@
-import React from 'react'
-import { connect } from 'react-redux'
-import { Table, Icon, Popup } from 'semantic-ui-react'
-
-import main from '../css/app-field.css'
+import React from 'react';
+import { connect } from 'react-redux';
+import { Table, Icon, Popup, Button, Label, Form, FormField } from 'semantic-ui-react';
+import main from '../css/app-field.css';
+import axios from 'axios';
+import { getCookie } from 'formula_one'
+import {urlAppHiddenDetail} from '../urls'
 
 class AppField extends React.PureComponent {
-  handleClick = () => {
-    let range = document.createRange()
-    range.selectNodeContents(this.textArea)
-    let sel = window.getSelection()
-    sel.removeAllRanges()
-    sel.addRange(range)
-    document.execCommand('copy')
-    sel.removeAllRanges()
+  constructor(props) {
+    super(props)
+    this.state = {
+      password_verified: this.props.password_verified,
+      formOpen : false,
+      password : "",
+      hiddenData : "",
+      message : ""
+    };
+  }
+  handleOpen = () => this.setState({ formOpen: true });
+
+  handleClose = () => this.setState({
+      formOpen: false,
+      message : "",
+      password : ""
+    });
+
+  handlePasswordChange = (e,{value}) => {
+    this.setState({
+      password : value
+    })
   }
-  render () {
-    const { activeApp, field, verboseName, editable } = this.props
-    const { data } = activeApp
+
+  handleSubmitSecret = (e) => {
+     e.preventDefault();
+
+     const headers = {
+      'X-CSRFToken': getCookie('csrftoken'),
+      'content-type' : 'application/json'
+    }
+
+     axios.post(urlAppHiddenDetail(),{
+      password : this.state.password,
+      id : this.props.activeApp.data['id']
+      },{headers: headers})
+      .then(res =>
+        {
+          this.setState({
+            password_verified : true,
+            hiddenData : res.data[this.props.field]
+          })
+        }
+        )
+        .catch(err => this.setState({message:err.response.data}))
+  }
+
+  handleClick = () => {
+    let range = document.createRange();
+    range.selectNodeContents(this.textArea);
+    let sel = window.getSelection();
+    sel.removeAllRanges();
+    sel.addRange(range);
+    document.execCommand("copy");
+    sel.removeAllRanges();
+  };
+  render() {
+    const {
+      activeApp,
+      field,
+      verboseName,
+      editable,
+    } = this.props;
+    const { data } = activeApp;
     return (
       <Table.Row>
         <Table.Cell>{verboseName}</Table.Cell>
+        {this.state.password_verified ? (
+          <Table.Cell>
+            <div styleName="desc-container">
+              <code ref={(textarea) => (this.textArea = textarea)}>
+                {this.state.hiddenData == ""
```

Comment:
Strings are enclosed in `''`
But **beware** - The values of JSX tags attributes are enclosed in `""`
```js
const name = 'Aman'
const component = <Table name="data" />
```

Reply:
For JSX, we generally follows <Table name={'data'} />. But genrally constants are not passed in the JSX elements. They are generally stae or props variables.